### PR TITLE
UC02-034: Fix `P_Next_Part_For_Decl` on generic declarations.

### PR DESCRIPTION
**Pull request by Roldak**

### CODE REVIEW

File: testsuite/tests/navigation/generic_pkg_top_level_2/foo.adb [hunk 1]
```diff
@@ -0,0 +1,3 @@
+package body Foo is
```
Comment (thvnx) [inline testsuite/tests/navigation/generic_pkg_top_level_2/foo.adb:1]: No test directives?

File: testsuite/tests/navigation/generic_pkg_top_level_2/foo.ads [hunk 2]
```diff
@@ -0,0 +1,5 @@
+generic
```
Comment (thvnx) [inline testsuite/tests/navigation/generic_pkg_top_level_2/foo.ads:1]: Same

File: testsuite/tests/navigation/generic_pkg_top_level_2/foo.adb (hunk 1 above)
Comment (thvnx) [reply testsuite/tests/navigation/generic_pkg_top_level_2/foo.adb:1]: Nevermind :D, that's good actually.

Comment (Roldak) [reply testsuite/tests/navigation/generic_pkg_top_level_2/foo.adb:1]: This is only to test the `p_body_part_for_decl` property, so the directive is in the `test.adb` file.

**Review by thvnx — APPROVED** [review]
LGTM

**Pull request merged**